app/Services/Attendance/AttendanceFactGenerationTracker.php

```
@@ -47,28 +49,46 @@ public function advanceBy(Employee $employee, CarbonInterface|string $workDate,
+            'updated_at' => now(),
+        ]);
+
+        $generation = AttendanceFactGeneration::withoutCompanyScope()
+            ->where('company_id', $employee->company_id)
+            ->where('employee_id', $employee->id)
+            ->whereDate('work_date', $date)
+            ->lockForUpdate()
+            ->firstOrFail();
 
-            $generation->increment('generation', $count);
+        $generation->increment('generation', $count);
```

PHPStan (2.2.7)

[error] Cannot call method increment() on stdClass.


(method.nonObject)